docs/analyzers/Using Additional Files.md

```diff
@@ -0,0 +1,231 @@
+```
+
+In a Project File
+=================
+
+Passing an Individual File
+--------------------------
+
+To specify an individual project item as an additional file, set the item type to `AdditionalFile`:
+
+``` XML
+<ItemGroup>
+  <AdditionalFile Include="terms.txt" />
+</ItemGroup>
+```
+
+Passing a Group of Files
+------------------------
+
+Sometimes it isn't possible to change the item type, or a whole set of items need to be passed as additional files. In this situation you can update the `AdditionalFileItemNames` property to specify which item types to include. For example, if your analyzer needs access to all .resx files in the project, you can do the following:
+``` XML
+<PropertyGroup>
+  <!-- Update the property to include all EmbeddedResource files -->
+  <AdditionalFileItemNames>$(AdditionalFileItemNames);EmbeddedResource</AdditionalFileItemNames>
+</PropertyGroup>
+<ItemGroup>
+  <!-- Existing resource file -->
+  <EmbeddedResource Include="Terms.resx">
+    ...
+  </EmbeddedResource>
+</ItemGroup>
+```
+
+Accessing Additional Files
+==========================
+
+The set of additional files can be accessed via the `Options` property of the context object passed to a diagnostic action. For example:
+```C#
+CompilationAnalysisContext context = ...;
+ImmutableArray<AdditionalText> additionFiles = context.Options.AdditionalFiles;
+```
+
+From an `AdditionalText` instance, you can access the path to the file or the contents as a `SourceText`:
+``` C#
+AdditionText additionalFile = ...;
+string path = additionalFile.Path;
+SourceText contents = additionaFile.GetText();
+```
+
+Samples
+=======
+
+Reading a File Line-by-Line
+---------------------------
+
+This sample reads a simple text file for a set of terms--one per line--that should not be used in type names.
+
+``` C#
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Text;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+public class CheckTermsAnalyzer : DiagnosticAnalyzer
+{
+    public const string DiagnosticId = "CheckTerms001";
+
+    private const string Title = "Type name contains invalid term";
+    private const string MessageFormat = "The term '{0}' is not allowed in a type name.";
+    private const string Category = "Policy";
+
+    private static DiagnosticDescriptor Rule =
+        new DiagnosticDescriptor(
+            DiagnosticId,
+            Title,
+            MessageFormat,
+            Category,
+            DiagnosticSeverity.Error,
+            isEnabledByDefault: true);
+
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get { return ImmutableArray.Create(Rule); } }
+
+    public override void Initialize(AnalysisContext context)
+    {
+        context.RegisterCompilationStartAction(compilationStartContext =>
```

Comment:
@ManishJayaswal I think that name is getting a little long, and being consistent with the type name doesn't add any clarity.